crates/consensus/src/transaction/mod.rs

```diff
@@ -51,6 +52,49 @@ pub trait Transaction: any::Any + Send + Sync + 'static {
     /// Get `gas_price`.
     fn gas_price(&self) -> Option<u128>;
 
+    /// Returns the EIP-1559 the maximum fee per gas the caller is willing to pay.
+    ///
+    /// For legacy transactions this is `gas_price`.
+    ///
+    /// This is also commonly referred to as the "Gas Fee Cap" (`GasFeeCap`).
+    fn max_fee_per_gas(&self) -> u128;
+
+    /// Returns the EIP-1559 Priority fee the caller is paying to the block author.
+    ///
+    /// This will return `None` for non-EIP1559 transactions
+    fn max_priority_fee_per_gas(&self) -> Option<u128>;
+
+    /// Return the max priority fee per gas if the transaction is an EIP-1559 transaction, and
+    /// otherwise return the gas price.
+    ///
+    /// # Warning
+    ///
+    /// This is different than the `max_priority_fee_per_gas` method, which returns `None` for
+    /// non-EIP-1559 transactions.
+    fn priority_fee_or_price(&self) -> u128;
```

Comment:
I think this can have a default impl like `self.max_priority_fee_per_gas().unwrap_or(self.max_fee_per_gas())`

though I'm wondering what's a usecase for this method?

Reply:
ah got it, its basically how execution layer treats legacy txses wrt priority fee